aspnetcore/security/cors.md

```diff
@@ -334,102 +346,239 @@ The response includes an `Access-Control-Allow-Methods` header that lists the al
 
 If the preflight request is denied, the app returns a *200 OK* response but doesn't send the CORS headers back. Therefore, the browser doesn't attempt the cross-origin request.
 
+-->
+
+<a name="apf"></a>
+
+### Automatic preflight request code
+
+When the CORS policy is applied either:
+
+* Globally by calling `app.UseCors("The Policy")` in `Startup.Configure`.
+* Using the `[EnableCors]` attribute.
+
+ASP.NET Core responds to the preflight OPTIONS request.
+
+Enabling CORS on a per-endpoint basis using `RequireCors` currently does ***not*** support automatic pre-flight requests.
+
+The [Test CORS](#testc) section of this document demonstrates this behavior.
+
+<a name="pro"></a>
+
+### [HttpOptions] attribute for preflight requests
+
+When CORS is enabled with the appropriate policy, ASP.NET Core generally automatically responds to CORS preflight requests. In some cases, code may not be generated to respond to preflight requests. For example, using [CORS with endpoint routing](#ecors).
```

Comment:
```suggestion
When CORS is enabled with the appropriate policy, ASP.NET Core generally responds to CORS preflight requests automatically. In some scenarios, this may not be the case. For example, using [CORS with endpoint routing](#ecors).
```